examples/abstract-objects/abstract-objects.bal

```diff
@@ -0,0 +1,57 @@
+import ballerina/io;
+
+// Defines an abstract object called 'Person'. It describes the type of
+// each field and method. However it cannot have a constructor method,
+// or any attached functions.
+type Person abstract object {
+    public int age;
+    public string firstName;
+    public string lastName;
+
+    // Function declarations can be within the object. But the function cannot 
+    // have a body.
+    function getFullName() returns string;
+
+    function checkAndModifyAge(int condition, int a);
+};
+
+// A non-abstract object called 'Employee', which is structurally equivalent
+// to 'Person'. It cannot have any member functions without a body.
```

Comment:
Minor changes:
"// Define a non-abstract object called 'Employee', which is structurally equivalent
// to 'Person'. Note that a non-abstract object cannot have any member functions without a body."